edge-agent/src/Microsoft.Azure.Devices.Edge.Agent.Service/modules/EdgeletModule.cs

```diff
@@ -67,6 +69,7 @@ public class EdgeletModule : Module
             this.productInfo = Preconditions.CheckNotNull(productInfo, nameof(productInfo));
             this.closeOnIdleTimeout = closeOnIdleTimeout;
             this.idleTimeout = idleTimeout;
+            this.performanceMetricsUpdateFrequency = performanceMetricsUpdateFrequency;
```

Comment:
nit: Philip has told me in the past to validate inputs at every level, however not having maxUpdateFrequency in scope means you can't do full validation. Is there a formal convention for input validation?

Reply:
I don't want to validate it at this level, b/c if metrics are disabled this value is ignored. I only want to validate it if it is used.